VideoLocker/src/main/java/org/edx/mobile/view/CourseOutlineFragment.java

```diff
@@ -141,34 +142,39 @@ public void updateMessageView(View view){
         }
     }
 
-    private void initializeAdapter(){
+    private void initializeAdapter() {
         if (adapter == null) {
             // creating adapter just once
-            adapter = new CourseOutlineAdapter(getActivity(), environment.getDatabase(), environment.getStorage(), new CourseOutlineAdapter.DownloadListener() {
+            adapter = new CourseOutlineAdapter(getActivity(), environment.getDatabase(),
+                    environment.getStorage(), new CourseOutlineAdapter.DownloadListener() {
                 @Override
                 public void download(List<HasDownloadEntry> models) {
-                    downloadManager.downloadVideos(
-                            (List) models, (FragmentActivity) getActivity(), (VideoDownloadHelper.DownloadManagerCallback) getActivity());
-
+                    CourseOutlineActivity activity = (CourseOutlineActivity) getActivity();
+                    if (NetworkUtil.isDownloadPossible(activity)) {
+                        downloadManager.downloadVideos(models, getActivity(),
+                                (VideoDownloadHelper.DownloadManagerCallback) getActivity());
+                    }
                 }
 
                 @Override
                 public void download(DownloadEntry videoData) {
-                    downloadManager.downloadVideo(
-                            videoData, (FragmentActivity) getActivity(), (VideoDownloadHelper.DownloadManagerCallback) getActivity());
+                    CourseOutlineActivity activity = (CourseOutlineActivity) getActivity();
+                    if (NetworkUtil.isDownloadPossible(activity)) {
+                        downloadManager.downloadVideo(videoData, activity, activity);
+                    }
+                }
+
+                @Override
+                public void viewDownloadsStatus() {
+                    environment.getRouter().showDownloads(getActivity());
                 }
             });
         }
 
-        if (!(NetworkUtil.isConnected(getActivity()))) {
-            AppConstants.offline_flag = true;
-        } else {
-            AppConstants.offline_flag = false;
-        }
+        AppConstants.offline_flag = !(NetworkUtil.isConnected(getActivity()));
```

Comment:
The parenthesis around the `(NetworkUtil.isConnected(getActivity())` call are unnecessary, and can be removed.